File: ros2_packages/hello_sub_py3/hello_sub_py3/hello_sub.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class HelloSub(Node):
    def __init__(self):
        super().__init__('hello_sub')
        topicName = '/hello'
        self.hello_msg = String()
        self.sub = self.create_subscription(String, 
                                            topicName, 
                                            self.helloCB, 
                                            10)
        self.sub

    def helloCB(self, msg):
        self.hello_msg = msg
        self.get_logger().info(msg)

def main(args=None):
    rclpy.init(args=None)
    hello_sub = HelloSub()
    try:
        rclpy.spin(hello_sub)
    except KeyboardInterrupt:
        print('killing process...')
    hello_sub.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: hello_ws/src/hello_test_sub_cpp/src/hello_sub.cpp
/* hello_sub */
#include <chrono>
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class HelloSub : public rclcpp::Node
{
public:
    using String = std_msgs::msg::String;

    explicit HelloSub(const rclcpp::NodeOptions &options);
private:
    String msg;
    rclcpp::Subscription<String>::SharedPtr sub;
private:
    void helloCB(const String::SharedPtr msg);
};
HelloSub::HelloSub(const rclcpp::NodeOptions &options=rclcpp::NodeOptions())
    : Node("hello_sub", options)
{
    using namespace std::placeholders;

    this->sub = this->create_subscription<String>(
        "/hello",
        10,
        std::bind(&HelloSub::helloCB, this, _1));
}
void HelloSub::helloCB(const String::SharedPtr msg)
{
    this->msg.data = msg->data;
    RCLCPP_INFO(this->get_logger(), "Received: %s", msg->data.c_str());
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto hello_sub = std::make_shared<HelloSub>();
    rclcpp::spin(hello_sub);

    rclcpp::shutdown();
}